Sources/SpeziSensorKit/BatchedAsyncDataFetcher.swift

```diff
@@ -0,0 +1,103 @@
+//
+// This source file is part of the SpeziSensorKit open source project
+//
+// SPDX-FileCopyrightText: 2025 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+@preconcurrency import SensorKit
+
+
+/// An `AsyncSequence` and `AsyncIterator` that can be used to fetch and process data from SensorKit, split into distinct batches.
+///
+/// Each batch will be fetched from SensorKit on demand, i.e. when ``next(isolation:)`` is called.
+///
+/// - Important: Due to the lazy nature of this type, and the fact that it uses a query anchor internally to keep track of already-fetched time ranhes,  the sequence should only be iterated once.
+@available(iOS 18, *)
+struct BatchedAsyncDataFetcher<Sample, SensorReader: SensorReaderProtocol<Sample>>: AsyncSequence, AsyncIteratorProtocol {
```

Comment:
we might want to call this `AnchoredAsyncDataFetcher` instead, to highlight the fact that it uses an anchor under the hood? (important bc it means that you can only iterate the sequence once, since the anchor will get updated in every iteration of a for loop over the sequence, meaning that you'll never get back to that previous state...)